src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -0,0 +1,617 @@
+
+    def _setup_manager(
+        self, kwargs
+    ) -> Tuple[Optional[ScheduledModifierManager], List[ScheduledModifierManager]]:
+        manager = None
+        arch_managers = []
+
+        if self.recipe is not None:
+            manager = ScheduledModifierManager.from_yaml(
+                self.recipe, recipe_variables=self.recipe_args
+            )
+            _LOGGER.info(
+                "Loaded SparseML recipe variable into manager for recipe: "
+                f"{self.recipe} and recipe_variables: {self.recipe_args}"
+            )
+
+        arch_recipe_paths = glob.glob(os.path.join(self.model_state_path, RECIPE_REGEX))
+        if arch_recipe_paths:
+            arch_managers = [
+                ScheduledModifierManager.from_yaml(path) for path in arch_recipe_paths
+            ]
+            _LOGGER.info(
+                f"Loaded SparseML {len(arch_recipe_paths)} recipes into architecture "
+                f"managers from {arch_recipe_paths}"
+            )
+
+        if manager is not None and manager in arch_managers:
+            # new recipe and the one stored with model are the same,
+            # keep manager and remove from arch_managers to keep from applying twice.
+            # remove this logic once recipe stages land
+            arch_managers.remove(manager)
+            _LOGGER.info(
+                "Removed duplicate SparseML recipe from arch_managers that matched "
+                "the recipe variable to prevent double application"
+            )
+
+        if (
+            manager is not None
+            and manager.max_epochs
+            and "args" in kwargs
+            and (hasattr(kwargs["args"], "num_train_epochs"))
+        ):
+            _LOGGER.warning(
+                f"Overriding num_train_epochs from Recipe to {manager.max_epochs}"
+            )
+            kwargs["args"].num_train_epochs = manager.max_epochs
+
+        return manager, arch_managers
+
+    def _reload_model_state(self, load_path: str, orig_state_dict: Dict[str, Any]):
+        if (
+            not load_path
+            or not os.path.isdir(load_path)
+            or not os.path.isfile(os.path.join(load_path, WEIGHTS_NAME))
+        ):
+            _LOGGER.warning(
+                "Model state was not reloaded for SparseML: "
+                f"could not find model wieghts for model_path {load_path}"
+            )
+            return
+
+        current_state_dict = self.model.state_dict()
+
+        if set(orig_state_dict.keys()) == set(current_state_dict):
+            # no change in keys, ignore reload
+            return
+
+        # change in keys due to architecture changes, reload statedict
+        load_state_dict = torch.load(
+            os.path.join(load_path, WEIGHTS_NAME), map_location="cpu"
+        )
+        _, missing, unexpected, __ = self.model._load_state_dict_into_model(
+            self.model, load_state_dict, load_path, _fast_init=False
+        )
+
+        if missing or unexpected:
```

Comment:
Can we separate these two warnings? I'm getting missing key found with an empty list.

Reply:
Yes, separated